src/controllers/QueryController.ts

```diff
@@ -580,16 +580,37 @@ export class QueryController extends RootComponent {
       return true;
     }
 
-    const enableHistory = this.searchInterface && this.searchInterface.options && this.searchInterface.options.enableHistory;
-    return enableHistory && this.getLastQueryHash() == this.queryHash(query, queryResults);
+    const enableHistory = !!(this.searchInterface && this.searchInterface.options && this.searchInterface.options.enableHistory);
+    return enableHistory && this.compareWithLastQueryHash(this.queryHash(query, queryResults));
+  }
+
+  private compareWithLastQueryHash(queryHash: string) {
+    const lastParams = JSON.parse(this.getLastQueryHash());
+    const newParams = JSON.parse(queryHash);
+    return Utils.objectEqual(lastParams, newParams);
   }
 
   private queryHash(query: IQuery, queryResults?: IQueryResults): string {
-    let queryHash = JSON.stringify(_.omit(query, 'firstResult', 'groupBy', 'debug'));
-    if (queryResults != null) {
-      queryHash += queryResults.pipeline;
+    const queryKeys: (keyof IQuery)[] = [
```

Comment:
One concern here is if a new key is added to `IQuery` that should be considered in the hash, would we be alerted to update this array?

Perhaps there is already an array or object in the code-base that lists what should be included in the hash? If so, keeping one source of truth would be useful,

Reply:
It's pretty unique, because it's omitting the page, number of results and all types of facets/groupBy, which all don't affect the searchUid.

There probably won't be any relevant keys added in the future, on the other hand, there was problematic keys added which weren't omitted. This is why i switched to a pick

Reply:
Yeah, I think that's the best decision if we assume that "not much will change" in JSUI: the odds of us supporting new query parameters is pretty low.